src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -432,26 +452,90 @@ record Element(Label target, Label next, Object caseLabel) {}
                         Label next = element.next();
                         cb.labelBinding(element.target());
                         if (element.caseLabel() instanceof Class<?> classLabel) {
-                            Optional<ClassDesc> classLabelConstableOpt = classLabel.describeConstable();
-                            if (classLabelConstableOpt.isPresent()) {
-                                cb.aload(0);
-                                cb.instanceof_(classLabelConstableOpt.orElseThrow());
-                                cb.ifeq(next);
+                            if (unconditionalExactnessCheck(selectorType, classLabel)) {
+                                //nothing - unconditionally use this case
+                            } else if (classLabel.isPrimitive()) {
+                                if (!selectorType.isPrimitive() && !Wrapper.isWrapperNumericOrBooleanType(selectorType)) {
+                                    // Object o = ...
+                                    // o instanceof Wrapped(float)
+                                    cb.aload(0);
+                                    cb.instanceof_(Wrapper.forBasicType(classLabel)
+                                                          .wrapperType()
+                                                          .describeConstable()
+                                                          .orElseThrow());
+                                    cb.ifeq(next);
+                                } else if (!unconditionalExactnessCheck(Wrapper.asPrimitiveType(selectorType), classLabel)) {
+                                    // Integer i = ... or int i = ...
+                                    // o instanceof float
+                                    Label notNumber = cb.newLabel();
+                                    cb.aload(0);
```

Comment:
we are wiring constants into code generation: `0`, `3` etc, probably those won't change but I would prefer using static final fields or enums, that can be documented and changed easily in the future if needed

Reply:
This code is intentionally using low-level primitives, as Classfile API is using pattern matching, and using some higher-level primitives is not possible due to the bootstrap issues. `parameterSlot` would probably work, but it does not seem too significant - we know the parameter types that are send to the method, and it is a static method.